### PR TITLE
Document that filesystem timestamps can be 0.

### PR DESCRIPTION
**Pull request by sunfishcode**
A value of 0 in a timestamp indicates the time is unknown. This ports #531 from Preview 1 to Preview 2.

### CODE REVIEW

**Review by pchickey — APPROVED** [review]

File: wit/types.wit [hunk 1]
```diff
@@ -110,10 +110,20 @@ interface types {
...
         /// Last data modification timestamp.
+        ///
+        /// A value of 0 indicates the platform doesn't maintain a
+        /// modification timestamp for this file.
+        data-modification-timestamp: datetime,
+        /// Last file status-change timestamp.
+        ///
+        /// A value of 0 indicates the platform doesn't maintain a
+        /// status-change timestamp for this file.
         data-modification-timestamp: datetime,
```
Comment (codefromthecrypt) [inline wit/types.wit:126]: looks like a bug?

```suggestion
        status-change-timestamp: datetime,
```

Comment (sunfishcode) [reply wit/types.wit]: Ah, this was a bug, but the bug was that I has copy-and-pasted an extra line. I've now fixed it.

**Comment by badeend** [comment]
Why did you choose a sentinel value over `option<datetime>` ?

**Comment by sunfishcode** [comment]
> Why did you choose a sentinel value over `option<datetime>` ?

Ah, I guess I was just thinking in terms of preview1. But you're right, `option` would be more appropriate here. I've now changed this.

**Pull request merged**